fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1707,6 +1718,89 @@ private List<Tuple> queryLocal(Map<Tuple, Long> dataset, Comparisons.Type compar
         return hits.stream().map(Map.Entry::getKey).collect(Collectors.toList());
     }
 
+    static Stream<Arguments> simplePartitionConsolidationTest() {
+        return Stream.of(
```

Comment:
Thinking about the testability, and complexity here, I wonder if it would make sense, and not be a big change, to have something like a `LuceneRepartitionPlanner` that takes a `List<LucenePartitionInfoProto.LucenePartitionInfo>` and returns a plan, which is something like:
> Move N newest documents from partition 3 to partition 4 and you are emptying partition 3.

Or

> Move M oldest documents from partition 2 to partition 1.

And then `LucenePartitioner` takes that plan and executes it in a single transaction. I'm not sure who is in charge of whether there is more to do.
That would probably allow decoupling the tests so you can have a lot of unit tests of the `RepartitionPlanner`, and fewer tests that actually involve the whole machinery and saving actual records to the database.